src/aws-redshiftserverless/workgroup.ts

```diff
@@ -0,0 +1,351 @@
+  }
+
+  /**
+   * The workgroup Arn
+   */
+  readonly workgroupArn: string;
+
+  /**
+   * The workgroup name
+   */
+  readonly workgroupName: string;
+
+  /**
+   * The workgroup id
+   */
+  readonly workgroupId: string;
+
+  /**
+   * The workgroup endpoint address
+   */
+  readonly endpointAddress: string;
+
+  /**
+   * The workgroup port
+   */
+  readonly port: number;
+
+  /**
+   * The connection object associated with the Redshift Serverless Workgroup.
+   */
+  public readonly connections: aws_ec2.Connections;
+
+  private readonly props: WorkgroupProps;
+  private readonly securityGroups: aws_ec2.ISecurityGroup[];
+  private readonly vpcSubnets: aws_ec2.SubnetSelection;
+
+  constructor(scope: Construct, id: string, props: WorkgroupProps) {
+    super(scope, id, {
+      physicalName:
+        props.workgroupName ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 64, allowedSpecialCharacters: '-' }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+
+    this.securityGroups = props.securityGroups ?? [this.createSecurityGroup()];
+
+    this.connections = this.createConnections();
+
+    this.vpcSubnets = props.vpcSubnets ?? {
+      subnetType: aws_ec2.SubnetType.PRIVATE_WITH_EGRESS,
+    };
+
+    this.validateCapacity();
+    this.validateWorkgroupName();
+    this.validatePort();
+    this.validateSubnet();
+
+    const workgroup = this.createWorkgroup();
+
+    this.workgroupArn = workgroup.attrWorkgroupWorkgroupArn;
+    this.workgroupName = workgroup.attrWorkgroupWorkgroupName;
+    this.workgroupId = workgroup.attrWorkgroupWorkgroupId;
+    this.endpointAddress = workgroup.attrWorkgroupEndpointAddress;
+    this.port = workgroup.attrWorkgroupEndpointPort;
+  }
+
+  protected createWorkgroup(): aws_redshiftserverless.CfnWorkgroup {
+    return new aws_redshiftserverless.CfnWorkgroup(this, 'Resource', {
+      baseCapacity: this.props.baseCapacity,
+      configParameters: this.props.configParameters
+        ? Object.entries(this.props.configParameters).map(([key, value]) => ({
+            parameterKey: key,
+            parameterValue: value,
+          }))
+        : undefined,
+      enhancedVpcRouting: this.props.enhancedVpcRouting,
+      namespaceName: this.props.namespace?.namespaceName,
+      publiclyAccessible: this.props.publiclyAccessible,
+      port: this.props.port,
+      securityGroupIds: Lazy.list({ produce: () => this.securityGroups.map(sg => sg.securityGroupId) }),
+      subnetIds: Lazy.list({ produce: () => this.props.vpc.selectSubnets(this.vpcSubnets).subnetIds }),
+      workgroupName: this.physicalName,
+    });
+  }
+
+  protected createSecurityGroup(): aws_ec2.SecurityGroup {
+    return new aws_ec2.SecurityGroup(this, 'SecurityGroup', {
+      vpc: this.props.vpc,
+      description: 'Automatic generated security group for Redshift Serverless Security Group',
+    });
+  }
+
+  protected createConnections(): aws_ec2.Connections {
```

Comment:
can be inlined

Reply:
I modified it in the same way as below.
https://github.com/open-constructs/aws-cdk-library/blob/main/src/aws-ec2/instance-connect-endpoint.ts#L92